persistence-modules/r2dbc/src/main/java/com/baeldung/examples/r2dbc/flyway/config/DatabaseConfig.java

```diff
@@ -0,0 +1,58 @@
+package com.baeldung.examples.r2dbc.flyway.config;
+
+import org.flywaydb.core.Flyway;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.data.r2dbc.config.AbstractR2dbcConfiguration;
+import org.springframework.data.r2dbc.repository.config.EnableR2dbcRepositories;
+
+import io.r2dbc.postgresql.PostgresqlConnectionConfiguration;
+import io.r2dbc.postgresql.PostgresqlConnectionFactory;
+import io.r2dbc.spi.ConnectionFactory;
+
+@Configuration
+@EnableR2dbcRepositories
+@Profile("!test")
+class DatabaseConfig extends AbstractR2dbcConfiguration {
+
+    @Value("${spring.r2dbc.host}")
+    private String host;
+    @Value("${spring.r2dbc.port}")
+    private Integer port;
+    @Value("${spring.r2dbc.database}")
+    private String database;
+    @Value("${spring.r2dbc.username}")
+    private String username;
+    @Value("${spring.r2dbc.password}")
+    private String password;
```

Comment:
sure, I understand your comment but I guess PostgresqlConnectionFactory doesn't accept an URL and R2dbcProperties doesn't have database/host, but still let me check if this can be achieved

Reply:
please ignore above comment, I am able to remove them by using URL, will update the PR